core/tests/style/pack/test_apply.py

```diff
@@ -72,3 +73,48 @@ def test_set_visibility_hidden():
     root = ExampleNode("app", style=Pack(visibility=HIDDEN))
     root.style.reapply()
     root._impl.set_hidden.assert_called_once_with(True)
+
+
+def test_set_visibility_inherited():
+    """Nodes should be hidden when an ancestor is hidden."""
+    grandparent = ExampleParentNode("grandparent", style=Pack())
+    parent = ExampleParentNode("parent", style=Pack())
+    child = ExampleNode("child", style=Pack())
+    grandparent.add(parent)
+    parent.add(child)
+
+    def assert_hidden_called(grandparent_value, parent_value, child_value):
```

Comment:
I suspect this convenience function may push the bounds of how fancy and smart a test should be, but I felt it was worth it for how much easier it made it to tell at a glance what's being tested at each step. Let me know if you'd like me to break it back out into individual checks.

Reply:
It's definitely pushing the bounds, but I think it's reasonable. It's an assertion that is specific to this particular test, but it's also a sequence of logic that would be tedious to invoke multiple times.